File: src/WorkBC.Data/Model/JobBoard/JobSeekerFlags.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;

namespace WorkBC.Data.Model.JobBoard
{
    public class JobSeekerFlags
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int Id { get; set; }

        [ForeignKey("JobSeeker")]
        public string AspNetUserId { get; set; }

        [JsonIgnore]
        public virtual JobSeeker JobSeeker { get; set; }

        public bool IsApprentice { get; set; }
        public bool IsIndigenousPerson { get; set; }
        public bool IsMatureWorker { get; set; }
        public bool IsNewImmigrant { get; set; }
        public bool IsPersonWithDisability { get; set; }
        public bool IsStudent { get; set; }
        public bool IsVeteran { get; set; }
        public bool IsVisibleMinority { get; set; }
        public bool IsYouth { get; set; }
    }
}

File: jobboard-laravel/app/Models/JobSeekerFlags.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

final class JobSeekerFlags extends Model
{
    protected $table = 'JobSeekerFlags';

    protected $primaryKey = 'Id';

    protected $keyType = 'int';

    public $incrementing = true;

    public $timestamps = false;

    protected $casts = [
        'IsApprentice' => 'boolean',
        'IsIndigenousPerson' => 'boolean',
        'IsMatureWorker' => 'boolean',
        'IsNewImmigrant' => 'boolean',
        'IsPersonWithDisability' => 'boolean',
        'IsStudent' => 'boolean',
        'IsVeteran' => 'boolean',
        'IsVisibleMinority' => 'boolean',
        'IsYouth' => 'boolean',
    ];

    public function jobSeeker(): BelongsTo
    {
        return $this->belongsTo(JobSeeker::class, 'AspNetUserId', 'Id');
    }
}
